x-pack/plugins/lens/public/editor_frame_service/format_column.ts

```diff
@@ -63,13 +69,27 @@ export const formatColumn: ExpressionFunctionDefinition<
       types: ['number'],
       help: '',
     },
+    nestedFormat: {
+      types: ['string'],
+      help: '',
+    },
+    template: {
+      types: ['string'],
+      help: '',
+    },
+    replaceInfinity: {
+      types: ['boolean'],
+      help: '',
+    },
```

Comment:
I'm thinking a little bit further ahead with these arguments, and I would prefer that we make slightly more generic arguments instead of specifically for ranges. The main reason is that later in this release cycle I'm expecting to add the time scaling formatters, which might have some custom arguments.

So usually when we want complex arguments, we use one of two approaches:

* Pass in a string of JSON which is parsed by the expression function
* Create a one-off expression function to hold the specific argument

In this case I prefer passing a JSON string, because that's similar to the way we define formatters in general.

Reply:
Cool, I was looking for a similar approach to avoid too many args, but couldn't find it in the codebase. Makes sense 👍 , I'll look now for that.